assets/js/device_manager.js

```diff
@@ -174,269 +185,105 @@
-      var tone_volume = 0;
-      var new_volume = 0;
-
-      switch(key) {
-        case "alerting":
-          tone_volume = this.Alerting.get_vol_step_factor();
-          break;
-        case "special":
-          tone_volume = this.SpecialTone.get_vol_step_factor();
-          break;
-        case "paging":
-          tone_volume = this.Paging.get_vol_step_factor();
-          break;
-        default:
-          console.error("Unknown transducer tone", key);
-          break;
-      }
-      if (level > 8) level = 8;
-      if (level < 1) level = 1;
-
-      // let new_volume = tone_volume;
-
-      if (level > 0) {
-        new_volume = level / tone_volume;
-      }
-
-      if (this.debug)
-        console.log('set_tone_volume', key, level, tone_volume, new_volume)
-
-      this.active_audio_ctrl.volume = new_volume;
-    },
-    transducer_tone_volume: function(msg) {
-      switch (msg.key) {
-        case "alerting":
-        case "special":
-          this.set_tone_volume(msg.tone_level, msg.key);
-          break;
-        default:
-          console.error('Unknown transducer tone', msg.key)
-          break;
-      }
+    set_volume_level: function(name, ...args) {
+      safe_call(this, name, 'set_volume_level', args);
     },
-    feedback: {
-      set_volume: function(value) {
-        console.log('TBD: implement this', value);
-      }
+    get_volume: function(name, ...args) {
+      safe_call(this, name, 'get_volume', args);
     },
-    Alerting: {
-      volume_up: function() {
-        console.log('TBD: implement this');
-      },
-      volume_down: function() {
-        console.log('TBD: implement this');
-      },
-      get_vol_step_factor: function() {
-        return 8;
-      }
+    set_tone_volume: function(name, ...args) {
+      safe_call(this, name, 'set_tone_volume', args);
     },
-    SpecialTone: {
-      get_vol_step_factor: function() {
-        return 8;
+*/
+    extend: function(name, extension, ...args) {
+      var extension = null;
+      if (extension = this.get_plugin(name)[extension]) {
+        extension(args);
       }
     },
-    Paging: {
-      get_vol_step_factor: function() {
-        return 8;
-      }
-    }
   };
 
+  $(document).ready(function() {
+  });
+
+  var DeviceManagerDefaultPlugin = {
+    is_device_manager: false,
+    init: function() {
+      console.log('The optional init callback');
+      // do so setup after the page has been loaded
+    },
+    start: function(args) {
+      this.is_device_manager = true;
+      console.log('starting device_manager with args', args)
+    },
+    stop: function(args) {
+      this.is_device_manager = false;
+      console.log('stopping device_manager with args', args)
+    },
+    volume_up: function() {},
+    volume_set: function(args) {
+      // note that calls with splat arguments (...args) come in as an array
+      var value = args[0];
+      $('#audio-control')[0].volume = value;
+    },
+    get_device: function(args) {},
```

Comment:
Should these not be implemented for the default adapter? Is this not the adapter we'll uses for WebRTC video?